### ISSUE TITLE
Deployment Scripts fail in AWS region ap-south-1

### ISSUE THREAD

**Issue by ChetanBhasin** [kubernetes/kubernetes#29627]
While using the deployment script, we get the following trace:

```
cluster/kube-up.sh 
... Starting cluster in ap-south-1a using provider aws
... calling verify-prereqs
... calling kube-up
Starting cluster using os distro: jessie
Uploading to Amazon S3
Creating kubernetes-staging-644ab343278c61ddae0ca23171f5546e
make_bucket failed: s3://kubernetes-staging-644ab343278c61ddae0ca23171f5546e/ An error occurred (InvalidRequest) when calling the CreateBucket operation: You are attempting to operate on a bucket in a region that requires Signature Version 4.  You can fix this issue by explicitly providing the correct region location using the --region argument, the AWS_DEFAULT_REGION environment variable, or the region variable in the AWS CLI configuration file.  You can get the bucket's location by running "aws s3api get-bucket-location --bucket BUCKET".
```

It looks like the AMIs are not available for this region. I got this trace after I copied the Kubernetes default AMI and to this region and setting `AWS_IMAGE` environment variable.

**Comment by oba11** [comment]
Also option `--cloud-provider="aws"` doesnt work for ap-south-1 as well. Not sure if the aws-sdk needs to be updated to reflect the new region

**Comment by oba11** [comment]
For my issue, https://github.com/kubernetes/kubernetes/pull/28428 fixes it but need to wait till v1.4

**Comment by justinsb** [comment]
I believe this is now fixed - please reopen if not
